### PR TITLE
Remedy deprecated dynamic variable declaration

### PR DESCRIPTION
**Pull request by alex-dna**
This should prevent warning to be triggered by php 8.2

### CODE REVIEW

File: src/Solr/Stores/SolrConfigStore_File.php [hunk 1]
```diff
@@ -49,4 +62,4 @@ public function instanceDir($index)
     {
         return $this->remote . '/' . $index;
     }
-}
+}
```
Comment (michalkleiner) [inline src/Solr/Stores/SolrConfigStore_File.php]: Please keep the new line at the end of the file, thanks

Comment (alex-dna) [reply src/Solr/Stores/SolrConfigStore_File.php]: done

**Review by GuySartorelli — CHANGES_REQUESTED** [review]
dynamic properties are public - these new properties must also be public to avoid backwards compatibility issues.

Please remove the default values - there are no default for dynamic properties, which these are replacing.

Please also target the `4.0` branch - ideally this change will be released as a patch release.

File: src/Solr/Stores/SolrConfigStore_File.php [hunk 2]
```diff
@@ -10,6 +10,19 @@
...
 {
+    /**
+     * @var string
+     */
+    protected $remote = '';
+
+    /**
+     * @var string
+     */
+    protected $local = '';
```
Comment (michalkleiner) [inline src/Solr/Stores/SolrConfigStore_File.php]:
```suggestion
    public $remote;

    /**
     * @var string
     */
    public $local;
```

File: src/Solr/Stores/SolrConfigStore_WebDAV.php [hunk 3]
```diff
@@ -12,6 +12,19 @@
...
 {
+    /**
+     * @var string
+     */
+    protected $remote = '';
+
+    /**
+     * @var string
+     */
+    protected $url = '';
```
Comment (michalkleiner) [inline src/Solr/Stores/SolrConfigStore_WebDAV.php]:
```suggestion
    public $remote;

    /**
     * @var string
     */
    public $url;
```

**Comment by michalkleiner** [comment]
Thought I might be able to batch commit the suggestions from GH UI, but no dice.

**Comment by GuySartorelli** [comment]
@alex-dna There are some requested changes on this PR - do you still want to have this change merged in?
If so, please make those changes. If the request changes aren't made in a few weeks I'll close this PR.

**Comment by alex-dna** [comment]
@GuySartorelli I've committed the requested changes. Thanks @michalkleiner . This is ow targetting the right branch I think.

**Review by GuySartorelli — APPROVED** [review]
Looks okay to me now. Thanks for that.

@michalkleiner do you have any concerns? (other than the merge commits in there, but I'll just squash-and-merge if you're happy with it)

**Review by michalkleiner — APPROVED** [review]
All good from my perspective, thanks @alex-dna and @GuySartorelli

**Pull request merged**